Entity/Components/StressComponent.gd

```diff
@@ -0,0 +1,122 @@
+class_name StressComponent extends EntityComponent
+## Holds entities stress
+##
+## Allows entities to deal stress damage or sooth. Manages stress generation
+
+## Emit when stress is changed
+signal on_stress_changed(new_stress: int)
+
+## The maximum stress the entity can have. This is a hard limit
+@export var max_stress: int = 100
+
+var default_stress: int
+
+## The amount of stress the entity generates at the start of each turn
+@export var stress_generation: int = 5
+
+## Know if an entity has been calmed already or not [br]
+## This is used to not distribute XP twice for the same entity, even if it is calmed multiple times [br]
+var has_been_calmed: bool = false
```

Comment:
Additionally, we are already checking for amount <= 0 in modify_stress, which should also include state of enemy already being calmed before - although having it checked via boolean would be more verbose I guess ?

Reply:
The placeholder function for XP is meant we I actually start implementing the XP so it's normal I didn't touch it. The `has_been_calmed` is used specifically for XP as described by the documentation comment, so it's normal I didn't use it yet.

I'm not sure what you mean by "which should also include state of enemy already being calmed before", could you be more explicit ? Linking to lines of code if possible. If you don't know how to do that, check [this github doc](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet)

Reply:
I get that XP is not implemented yet, but since you already added a placeholder function for it as well as bool to check it, I don't see why not also add the check for enemy being calmed that would call said function. It's not a big issue though, it can be added later in other PR

What I meant was in modify_stress, [you are not doing anything if stress <= 0](https://github.com/Saplings-Projects/1M_sub/blob/98a828d80665699405b934670d0e24feed957191/Entity/Components/StressComponent.gd#L58-L59), which means if enemy was calmed, their stress is 0 and that check effectively serves the same purpose as the "has_been_calmed" bool I believe. This is based on my understanding that XP handling would be called from modify_stress however, maybe I assumed wrong and in that case there might be a use for it

Reply:
My bad on the amount one, misread that the first time and then just had it's meaning mistaken. In that case yes, it's different.

_"because a mob that has been soothed can still be stressed again after"_
Is that the case ? Currently if stress reaches 0 enemies stop generating it no ? [(because of on_turn_start)](https://github.com/Saplings-Projects/1M_sub/blob/98a828d80665699405b934670d0e24feed957191/Entity/Components/StressComponent.gd#L93)
Did we decide if we want once soothed enemies to still be gaining stress ?

Reply:
They stop generating stress indeed. But the player could use a card the stresses the enemy (example, the one that deals 10 damage and 10 stress). Meaning enemies would go back to above 0 and they would start generating stress again.
But naturally, they will stay soothed if you don't stress them more.